File: ios/SwiftKotlination/SwiftKotlination/Sources/Features/Story/Model/StoryManager.swift
protocol StoryManagerProtocol {
    func story(_ observer: @escaping Observer<Story>)
}

struct StoryManager {
    let story: Story
}

extension StoryManager: StoryManagerProtocol {
    func story(_ observer: @escaping Observer<Story>) {
        observer(.success(story))
    }
}

File: android/SwiftKotlination/app/src/main/java/fr/jhandguy/swiftkotlination/features/story/model/StoryManager.kt
package fr.jhandguy.swiftkotlination.features.story.model

import fr.jhandguy.swiftkotlination.observer.Observer
import fr.jhandguy.swiftkotlination.observer.Result

interface StoryManagerInterface {
    suspend fun story(observer: Observer<Story>)
}

class StoryManager(private val story: Story): StoryManagerInterface {
    override suspend fun story(observer: Observer<Story>) = observer(Result.Success(story))
}